internal/services/network/subnet_resource.go

```diff
@@ -27,6 +27,7 @@ var SubnetResourceName = "azurerm_subnet"
 var subnetDelegationServiceNames = []string{
 	"Microsoft.ApiManagement/service",
 	"Microsoft.AzureCosmosDB/clusters",
+	"Microsoft.BareMetal/AzureHostedService",
```

Comment:
could we also add this to the list of possible values in the docs?

Reply:
sure, I've updated the document  as well